dfusion_rust_core/src/models/standing_order.rs

```diff
@@ -123,31 +128,67 @@ impl From<&Arc<Log>> for StandingOrder {
 }
 
 
-impl From<Entity> for StandingOrder {
-    fn from(entity: Entity) -> Self {
-        let account_id = u16::from_entity(&entity, "accountId");
-        let batch_index = U256::from_entity(&entity, "batchIndex");
-        let valid_from_auction_id = U256::from_entity(&entity, "validFromAuctionId");
-
-        StandingOrder {
-            account_id,
-            batch_index,
-            valid_from_auction_id,            
-            orders: vec![] // Orders cannot be reconstructed just from the entity
+impl From<(Entity, Vec<Entity>)> for StandingOrder {
+    fn from(entities: (Entity, Vec<Entity>)) -> Self {
+        let batch_entity = entities.0;
+        let mut order_entities_iter = entities.1.into_iter();
+        let order_ids_value = batch_entity.get("orders").expect("it should contain order ids");
+
+        if let Value::List(order_ids) = order_ids_value {
+            assert_eq!(
+                order_ids.len(),
+                order_entities_iter.len(),
+                "The entity should have the same number of orders as Vec<Entity>"
+            );
+
+            let account_id = u16::from_entity(&batch_entity, "accountId");
+            let batch_index = U256::from_entity(&batch_entity, "batchIndex");
+            let valid_from_auction_id = U256::from_entity(&batch_entity, "validFromAuctionId");
+
+            let orders = order_ids.iter()
+                .map(|order_id| {
+                    // Get matching order for the id
+                    let order_entity: Entity = order_entities_iter
+                        .find(|current_order| {
+                            let current_order_id = current_order.get("id")
+                                .expect("The entity orders should have an id that match the standing order");
+                                
+                            current_order_id == order_id
+                        })
+                        .expect("The standing order has and id not found in the entity Vec");
+
+                    // Convert order into Entity
+                    order_entity
+                })
+                .map(|a| a.into())
+                .collect();
+
+            StandingOrder {
+                account_id,
+                batch_index,
+                valid_from_auction_id,
+                orders
+            }
+        } else {
+            panic!("orders should be a List")
         }
     }
 }
 
 impl Into<Entity> for StandingOrder {
     fn into(self) -> Entity {
-        let mut entity = Entity::new();
-                
+        let mut entity = Entity::new();                
         entity.set("accountId", self.account_id.to_value());
         entity.set("batchIndex", self.batch_index.to_value());
         entity.set("validFromAuctionId", self.valid_from_auction_id.to_value());
+
+        // let orders = standing_order.get_orders().clone();
```

Comment:
Should we remove and replace with a comment/task?

Reply:
ups..